client/src/components/Upload.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <v-card width="600" height="160" raised class="mx-auto">
+    <v-card-title>Upload spreadsheet</v-card-title>
+    <br>
+    <v-card-text>
+      <v-file-input accept="application/vnd.ms-excel, application/msexcel, application/x-msexcel, application/x-ms-excel, 
+    application/x-excel, application/x-dos_ms_excel, application/xls, application/x-xls,
+    application/vnd.openxmlformats-officedocument.spreadsheetml.sheet" label="Click here to select a file to upload"
+        outlined @change="uploadFile" show-size placeholder="Pick a file to upload" >
+      </v-file-input>
+    </v-card-text>
+  </v-card>
+</template>
+<script lang="ts">
+import { read, utils } from "xlsx"
+import BenefitMatrix, { Header, TariffDetails } from "./BenefitMatrix"
+
+export default {
+  methods: {
+    /**
+     * Currently, this function isn't actually uploading, just parsing.
+     * I assume this will be done on the server eventually.
+     * @param event 
+     */
+    uploadFile(event) {
+      var chosenFile = event.target.files[0];
+      const reader = new FileReader();
+
+      reader.onload = (progressEvent) => {
+        if (progressEvent === null) {
+          return
+        }
+
+        // Parse data 
+        const bstr = progressEvent.target.result
```

Comment:
I would like to move the parsing logic into a different function, which shouldn't be available in the `template` scope. Is there a way to do that?